luigi/scheduler.py

```diff
@@ -691,6 +717,18 @@ def update_resources(self, **resources):
             self._resources = {}
         self._resources.update(resources)
 
+    def _get_task_level_config(self, config):
+        if config:
+            return {
+                'disable_failures': config.get('disable-num-failures', self._config.disable_failures),
```

Comment:
I would guess this could all be simplified and this conversion-code made go away. Also consider packing things in a named-tuple. Just an idea :)